@here/harp-mapview/lib/Utils.ts

```diff
@@ -266,12 +266,16 @@ export namespace MapViewUtils {
         }
         const mapTargetNormal = projection.surfaceNormal(mapTargetWorld, new THREE.Vector3());
 
-        const dirMapTarget = camera.position
-            .clone()
-            .sub(mapTargetWorld)
-            .normalize();
+        camera.updateMatrixWorld(true);
+        const cm = camera.matrixWorld.elements;
+        const dir = cache.vector3[1].set(cm[8], cm[9], cm[10]);
```

Comment:
Consider `camera.getWorldDirection`, or `extractBasis` from the `matrixWorld` ?

Reply:
I actually copied the code from camera.getWorldDirection to avoid an extra updateWorldMatrix call.
Maybe `camera.matrixWorld.setFromMatrixColumn` is more readable

Reply:
Sounds good.